src/udev/udevadm-info.c

```diff
@@ -255,6 +255,60 @@ static void cleanup_dir(DIR *dir, mode_t mask, int depth) {
         }
 }
 
+/*
+ * Assume that dir is a directory with file names matching udev data base
+ * entries for devices in /run/udev/data (such as "b8:16"), and removes
+ * all files except those that haven't been deleted in /run/udev/data
+ * (i.e. they were skipped during db cleanup because of the db_persist flag).
+ * Returns true if the directory is empty after cleanup.
+ */
+static bool cleanup_dir_after_db_cleanup(DIR *dir, DIR *datadir) {
+        unsigned int kept = 0;
+
+        assert(dir != NULL && datadir != NULL);
```

Comment:
Not a big matter, but we accept implicit cast from pointer to boolean, hence `!= NULL` is not necessary.
Also, that's super minor comment, but we usually add one empty line between assertions and other parts:
```
static bool cleanup_dir_after_db_cleanup(DIR *dir, DIR *datadir) {
        unsigned int kept = 0;

        assert(dir);
        assert(datadir);

        FOREACH_DIRENT_ALL(...) {
                ...
        }
```